.github/WORKFLOW_DISPATCH.md

```diff
@@ -0,0 +1,131 @@
+
+```yaml
+jobs:
+  say_hello:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Say hello customer
+      run: |
+        MANUAL_TRIGGER_CUSTOMER="${{ github.event.inputs.customer}}"
+        echo "Hello $MANUAL_TRIGGER_CUSTOMER"
+```
+
+All inputs are available within `${{github}}` variable. Because this input has been explicitly required using `required: true`, the `${{ github.event.inputs.customer}}` value cannot be empty.
+
+In order to use a **fallback** value (useful when the input isn't required, e.g: `required: false`), we can use this bash trick:
+
+```bash
+CUSTOMER_TO_DEPLOY="${MANUAL_TRIGGER_CUSTOMER:-$(cat now.json | jq -r '.build.env.NEXT_PUBLIC_CUSTOMER_REF')}"
+```
+
+If `MANUAL_TRIGGER_CUSTOMER` is empty, then we'll resolve its fallback value from the `now.json` file and extract the `build.env.NEXT_PUBLIC_CUSTOMER_REF` value.
+
+## 3. How to trigger the workflow using an external HTTP request?
+
+### Authentication
+
+Authentication is required for every request, you'll need to provide your credentials through an HTTP header:
+
+```json
+{
+    "Accept": "application/vnd.github.v3+json",
+    "Authorization": "token <TOKEN>"
+}
+```
+
+**N.B**: This `token` needs the **full repo** permissions to be granted access.
+
+### Get your workflow id
+
+#### Why do we need it?
+
+Each GitHub workflow has an id, and this id changes for every repository.
+
+**You'll need this workflow id to be able to trigger the workflow from an external HTTP request.**
+
+#### How can we get it?
+
+```
+GET https://api.github.com/repos/{owner}/{repo}/actions/workflows
+```
+
+ - **`owner`**: The github nickname/organization
+ - **`repository-name`**: The repository name where to trigger the action
+
+This call will return all your workflow, so make sure you use the right one.
+
+### Example
+
+This `curl` command will sort you the ID, if you have not change the *name/path*. Please change the **owner** and the **repository name**.
+
+```bash
+curl -s \
+  -X GET \
+  -H "Accept: application/vnd.github.v3+json" \
+  -H "Authorization: token <YOUR_GITHUB_TOKEN>" \
+  https://api.github.com/repos/UnlyEd/next-right-now/actions/workflows | jq '.workflows[] | select(.path==".github/workflows/deploy-zeit-staging.yml") | .id'
+```
+
+### Structure
+
+#### Endpoint
+
+The endpoint: `POST https://api.github.com/repos/{owner}/{repository-name}/actions/workflows/{workflow-id}/dispatches`
+
+ - **`owner`**: the github nickname/organization
+ - **`repository-name`**: the repository name to trigger the action
+ - **`workflow-id`**: the workflow id (check the previous section to learn how to get it)
+
+#### Body
+
+You have to provide a body with these inputs:
+
+```json
+{
+    "ref": "master",
+    "inputs": {
+        "customer": "customer2"
+    }
+}
+```
+
+ - **ref**: a git **reference**, it can be a tag, a branch or a SHA commit. Can be really useful to deploy a customer using a particular version instead of the master branch.
```

Comment:
I don't understand how the ref is being used in our script, will it really trigger a Vercel deployment using the ref provided to the action? Is it because the "ref" will target a different commit and run the worklow on a different commit, which in turn will trigger the Vercel deployment based on a different source code? 🤔

Reply:
@Vadorequest *ref* allows you to specify a commit, tag, branch where the actions will be run :)
So the version code you want to deploy !